racket/commands/macro.rkt

```diff
@@ -158,7 +158,9 @@
   (define after-file  (make-temporary-file-with-text after-text))
   (define out (open-output-string))
   (begin0 (parameterize ([current-output-port out])
-            (system (format "diff -U ~a ~a ~a" -U before-file after-file))
+            (if (eq? (system-type) 'windows)
+                (system (format "diff --strip-trailing-cr -U ~a ~a ~a" -U before-file after-file))
+                (system (format "diff -U ~a ~a ~a" -U before-file after-file)))
             (match (get-output-string out)
```

Comment:
Another idea would be to change the result string ourselves. For example here do something like `(match (regexp-replace* #rx"\r\n" (get-output-string out) "\n")`.

I think it's probably fine to do this unconditionally (it's harmless when not on Windows)?

Reply:
I have tried your approach, it is OK. I also think it should be harmless.